src/memory/cgroup_v2.rs

```diff
@@ -0,0 +1,163 @@
+#[cfg(target_os = "linux")]
+pub mod os {
+    use std::io::BufRead;
+    use std::path::Path;
+
+    use super::super::Memory;
+    use crate::{file_to_buf_reader, parse_u64};
+    use crate::{path_to_string, ProbeError, Result};
+
+    pub fn read_and_parse_v2_sys_memory(path: &Path) -> Result<Memory> {
+        let mut memory = Memory {
+            total: None,
+            free: None,
+            used: 0,
+            buffers: 0,
+            cached: 0,
```

Comment:
I didn't make them optional before because they weren't accessed publicly, they only really exist for the test assertions. I could make them optional, conceptually it's the right way to express this. 👍